### ISSUE TITLE
[Bug]: renderFormIdInput (in ValidatedForm) stops server-side field validation errors from showing up

### ISSUE THREAD

**Issue by ttiurani** [airjp73/rvf#397]
### Which packages are impacted?

- [ ] `@rvf/react`
- [X] `@rvf/remix`
- [X] `@rvf/zod`
- [ ] `@rvf/yup`
- [ ] `zod-form-data`

### What version of these packages are you using?

- `@rvf/remix`: 6.2.0
- `@rvf/zod`: 6.0.3

### Please provide a link to a minimal reproduction of the issue.

See below

### Steps to Reproduce the Bug or Issue

Using this kind of client validation:

```tsx
const nonEmpty = (message: string) =>
    z
        .string()
        .transform((t) => t?.trim())
        .pipe(z.string().min(1, { message }));
export const debugSchema = z.object({
    debug: z.string().pipe(nonEmpty("Value cannot be empty")),
});
export const debugValidator = withZod(debugSchema);
```

and this kind of server validation:

```tsx
export const getServerDebugValidator = () =>
    withZod(
        debugSchema.transform(async (val, ctx) => {
            if (val.debug === "debug") {
                ctx.addIssue({
                    code: z.ZodIssueCode.custom,
                    message: "SERVER DEBUG ERROR",
                    path: ["debug"],
                });
                return z.NEVER;
            }
        }),
    );
```

and this kind of action:

```tsx
export async function action({ request }: ActionFunctionArgs) {
    const result = await getServerDebugValidator().validate(await request.formData());
    if (result.error) return validationError(result.error);
    return result;
 }
```

then this form shows correctly on the client the "Value cannot be empty" validation, but when submit is pressed, nothing shows up:

```tsx
    const form = useForm({
        defaultValues: { debug: "debug" },
        validator: debugValidator,
        method: "post",
    });
    return (
        <form {...form.getFormProps()}>
            {form.renderFormIdInput() /* <- If that is commented out the SERVER DEBUG ERROR shows up */}
            <input data-testid="debug" {...form.getInputProps("debug")} />
            {form.error("debug") && <pre data-testid="foo-error">{form.error("debug")}</pre>}
            <button type="submit" data-testid="submit">
                submit
            </button>
        </form>
    );
```

The error only starts to show up when `form.renderFormIdInput()` is commented out, but then the server action response also has `formId: undefined`. That likely means multiple forms on one page does not work – however, I didn't test multiple forms so this is just guessing.

This problem also manifests itself with `ValidatedForm`. With it too "SERVER DEBUG ERROR" does not show up, unless [this line](https://github.com/airjp73/rvf/blob/4d8da5b9ce0674ddee03a4f23a7993854f7d37eb/packages/remix/src/ValidatedForm.tsx#L93) is commented out, after which it works.

### Expected behavior

The server validated field errors should show up.

### Platform

- OS: macOS
- Browser: any

**Comment by airjp73** [comment]
Hi! In your code example you're using `renderFormIdInput`, but you haven't passed an `id` to `useForm`. Does that fix the issue?

**Comment by airjp73** [comment]
I guess it's not explicitly mentioned, but it's in the code example here: https://www.rvf-js.io/remix#caveats

With the generated id, if you don't have JS enabled or it isn't loaded yet, the page will get server rendered again after your action is done, which will generate a new id. So the id from useId will no longer be consistent with what's returned from the server.

**Comment by ttiurani** [comment]
> I guess it's not explicitly mentioned, but it's in the code example here: https://www.rvf-js.io/remix#caveats

I actually did read that caveat, but it says "If you have more than one form active on the page..", and I have only one form.

Currentlt the id is mandatory if you want to ever do any server-side validation, which I'd think is the majority of use caees (all of my use cases need this feature).

For that readon I'd say it's important enough that I'd definitely put the `id` in every single `useForm` and `ValidatedForm` documentation example and make the "no id" case a caveat 

However, I still don't understand why in 99% of the cases where JS is enabled, the generated forn id can't be made to match. The HTML has `<form id=":r1:`>` and the server returns that id, so it should not be impossible?

**Comment by airjp73** [comment]
> Currently the id is mandatory if you want to ever do any server-side validation

If that's the case then it's definitely a bug. Having no id should be supported when there's only one form on the page.

It does look like there might be something wonky going on when I look at the code handling this feature. Would you be able to provide a reproduction in a repo or code sandbox? Or better yet if you have time to contribute a test case [in this file](https://github.com/airjp73/rvf/blob/main/packages/remix/src/test/submission.test.tsx) that would be amazing.

> However, I still don't understand why in 99% of the cases where JS is enabled, the generated forn id can't be made to match. The HTML has <form id=":r1:>` and the server returns that id, so it should not be impossible?

You already have to explicitly add `renderFormIdInput` to handle server-side validation. Adding 2 lines of code that covers 100% of cases is better than adding 1 line of code that covers 99% of cases IMO.

**Comment by ttiurani** [comment]
> > Currently the id is mandatory if you want to ever do any server-side validation
> 
> If that's the case then it's definitely a bug. Having no id should be supported when there's only one form on the page.

A clarification here, my statement was a little to sweeping: from what I tried the `id` seems to mandatory for `ValidatedForm` because it always has inside it the call to `renderFormIdInput()`, which causes this problem.

`useForm` without `id` **and** without `renderFormIdInput()` works with server returning `formId: undefined` and then the form is found (presumably) somehow by just hitting the first form?

So the issue is more pronounced with `<ValidatedForm>`.

**Comment by airjp73** [comment]
Yeah, that make sense. `renderFormIdInput` has built-in logic to not render when there's no formId, but I think that's what's broken.

**Comment by airjp73** [comment]
Should be fixed in 6.3. Please feel free to re-open if it's still broken.
